protocols/messages.go

```diff
@@ -201,3 +202,50 @@ func (m Message) Summarize() MessageSummary {
 	}
 	return s
 }
+
+// These types are used to implement the zerolog.LogObjectMarshaler interface
+type RejectedIds []string
+type ObjectivePayloadSummaries []ObjectivePayloadSummary
+type ProposalSummaries []ProposalSummary
+type PaymentSummaries []PaymentSummary
+
+func (m MessageSummary) MarshalZerologObject(e *zerolog.Event) {
+	e.Str("To", m.To).
+		Array("PayloadSummaries", ObjectivePayloadSummaries(m.PayloadSummaries)).
+		Array("ProposalSummaries", ProposalSummaries(m.ProposalSummaries)).
+		Array("Payments", PaymentSummaries(m.Payments)).
+		Array("RejectedObjectives", RejectedIds(m.RejectedObjectives))
+
+}
+func (o ObjectivePayloadSummaries) MarshalZerologArray(a *zerolog.Array) {
+	for _, p := range o {
+		a.Object(p)
+	}
+}
+
+func (o ProposalSummaries) MarshalZerologArray(a *zerolog.Array) {
+	for _, p := range o {
+		a.Object(p)
+	}
+}
+
+func (o PaymentSummaries) MarshalZerologArray(a *zerolog.Array) {
+	for _, p := range o {
+		a.Object(p)
+	}
+}
```

Comment:
Nit: these all look the same. Any way to factor this out?

Reply:
I've used some generics to clean this up in https://github.com/statechannels/go-nitro/pull/1112/commits/33948e8764038d33c30557e44b7ba4075ab66e9a